github/release_notes/renderer.py

```diff
@@ -121,93 +123,102 @@ def _nodes_to_lines(
-                else:
-                    if not self._generate_link(rn_obj):
-                        header_suffix_list.append(
-                            '{repo_path}{reference}'.format(
-                                repo_path=cn.github_repo_path(),
-                                reference=reference
-                            )
-                        )
-                    else:
-                        if rn_obj.reference_is_pr:
-                            reference_link = '{source_repo_url}/pull/{ref_id}'.format(
-                                source_repo_url=rn_obj.cn_source_repo.github_repo_url(),
-                                ref_id=rn_obj.reference_id
-                            )
-                        else: # commit
-                            reference_link = '{source_repo_url}/commit/{ref_id}'.format(
-                                source_repo_url=rn_obj.cn_source_repo.github_repo_url(),
-                                ref_id=rn_obj.reference_id
-                        )
+    def _header_suffix_reference(
+        self,
+        rls_note_obj: ReleaseNote
+    ):
+        cn_source_repo = rls_note_obj.cn_source_repo
+        reference_id_text = rls_note_obj.reference_id
 
-                        link_text = '{repo_path}{reference}'.format(
-                            repo_path=cn.github_repo_path(),
-                            reference=reference
-                        )
-                        header_suffix_list.append(
-                            self._build_link(url=reference_link, text=link_text)
-                        )
-            if rn_obj.user_login:
-                if not self._generate_link(rn_obj):
-                    header_suffix_list.append('@{u}'.format(
-                        u=rn_obj.user_login
-                    ))
-                else:
-                    user_link_text = '@{u}'.format(u=rn_obj.user_login)
-                    user_url = '{github_url}/{u}'.format(
-                         u=rn_obj.user_login,
-                        github_url=cn.github_url()
+        if rls_note_obj.reference_is_pr:
+            reference_prefix = '#'
+        else: # commit
+            if not rls_note_obj.is_current_repo or self._generate_link(rls_note_obj):
+                reference_prefix = '@'
+            else:
+                # for the current repo we use gitHub's feature to auto-link to references,
+                # hence in case of commits we don't need a prefix
+                reference_prefix = ''
+            if self._generate_link(rls_note_obj):
+                reference_id_text = rls_note_obj.reference_id[0:12] # short commit hash
+
+        reference = '{reference_prefix}{ref_id}'.format(
+            reference_prefix=reference_prefix,
+            ref_id=reference_id_text,
+        )
+
+        if rls_note_obj.is_current_repo and not self._generate_link(rls_note_obj):
+            return reference
+        else:
```

Comment:
rm the else and dedent the following coding, please (that's the whole point of the early exit, is it not?) :-)

Reply:
absolutely 👍